docs/modules/ROOT/pages/how-tos/workflows/git-submodules.adoc

```diff
@@ -0,0 +1,109 @@
+
+- **Upstream repository**: a community project which you do not have push access to (or which you do not want to push CI changes into).
+
+- **Downstream repository**: a self-owned copy of the upstream repository whose build, test, release lifecycle you want to control completely.
+
+=== Create a downstream repository
+
+The downstream repository can be used to do the following:
+
+* hold the git submodule reference to the upstream repository.
+* use the contents of the git submodule in a Containerfile for further edits.
+* onboarded onto {ProductName}.
+
+==== Configuring components
+
+CAUTION: Make sure to properly set the context and Containerfile path as required for the artifact builds.
+You can keep the context value at the the repository root where the Containerfile and git submodules are located.
+
+See xref:/how-tos/creating.adoc[Creating applications and components] for more details.
+
+==== Configuring the Containerfile
+
+In the Containerfile you can specify the git submodule directories and add any necessary edits that differ from the upstream repository. Here is a basic example:
+
+[source, Dockerfile]
+----
+FROM <base-image>
+
+# Install necessary dependencies
+RUN dnf -y install <dependencies>
+
+# Set the working directory
+ENV <SUBMODULEPKG> /path/to/submodule
+WORKDIR ${<SUBMODULEPKG>}
+
+# Commands to update the submodule for the current image build
+RUN sed -i 's/<old-text>/<new-text>/g' file/in/submodule/directory
+# Commands to build binary from submodule sources
+RUN go build file/in/submodule/directory
+
+WORKDIR /app
+
+LABEL name="<name" \
+      summary="<summary>" \
+      description="<description>"
+
+ENTRYPOINT ["<entrypoint-executable>"]
+----
+
+=== Create the git submodule for the upstream repository to track it
+
+After creating the downstream repository, add a git submodule for the upstream repository:
+
+[source, bash]
+----
+git submodule add <upstream-url>
+----
+
+Check and configure the git submodule config:
+[source, gitmodules]
+----
+# .gitmodules
+[submodule "<project-name>"]
+  path = <path-in-downstream-repository>
+  url = https://github.com/<namespace>/<project>.git
+  # Please note: by default a branch is not specified
+  # in the git submodule configuration file.
+  # Please consider setting one to prevent using the wrong branch.
+  # Here is an example of a branch being set:
+  branch = <designated-repository-branch>
+----
+
+See the link:https://git-scm.com/docs/gitsubmodules[Git documentation] for more details on configuring git submodules.
+
+=== Onboard the component onto {ProductName}
+
+After creating the downstream repository you can onboard the component. See xref:/how-tos/creating.adoc[Creating applications and components] for more details.
+
+=== Using private repositories as a git submodule:
+
+When using private repositories we encourage that you give the link:https://github.com/apps/red-hat-konflux[Red Hat {ProductName} Github app] access to that private repository.
```

Comment:
This is a GitHub app for our instance and is not a general purpose GitHub app.

Reply:
Hmm... should I move this to the downstream docs then?

Reply:
I removed this section, see https://github.com/konflux-ci/docs/pull/120/commits/fa1880375b27566d5555d7851d224d82d735605e